iroha_client_cli/src/main.rs

```diff
@@ -0,0 +1,307 @@
+                .long(CONFIG)
+                .value_name("FILE")
+                .help("Sets a config file path.")
+                .takes_value(true)
+                .default_value("config.json"),
+        )
+        .subcommand(
+            App::new(CREATE)
+                .about("Use this command to request entities creation in Iroha Peer.")
+                .subcommand(
+                    domain::build_create_app(),
+                )
+                .subcommand(
+                    account::build_create_app(),
+                )
+                .subcommand(
+                    asset::build_create_app(),
+                )
+        )
+        .subcommand(
+            App::new(UPDATE)
+            .about("Use this command to request entities related modifications in Iroha Peer.")
+            .subcommand(
+                asset::build_update_app(),
+            )
+        )
+        .subcommand(
+            App::new(GET)
+            .about("Use this command to request entities related information from Iroha Peer.")
+            .subcommand(
+                asset::build_get_app(),
+            )
+        )
+        .get_matches();
+    if let Some(configuration_path) = matches.value_of(CONFIG) {
+        println!("Value for config: {}", configuration_path);
+    }
+    if let Some(ref matches) = matches.subcommand_matches(CREATE) {
+        domain::match_create(matches);
+        account::match_create(matches);
+        asset::match_create(matches);
+    }
+    if let Some(ref matches) = matches.subcommand_matches(UPDATE) {
+        asset::match_update(matches);
+    }
+    if let Some(ref matches) = matches.subcommand_matches(GET) {
+        asset::match_get(matches);
+    }
+}
+
+mod domain {
+    use super::*;
+    use clap::ArgMatches;
+    use futures::executor;
+    use iroha::{domain::isi::CreateDomain, prelude::*};
+    use iroha_client::client::Client;
+
+    const DOMAIN: &str = "domain";
+    const DOMAIN_NAME: &str = "name";
+
+    pub fn build_create_app<'a, 'b>() -> App<'a, 'b> {
+        App::new(DOMAIN).arg(
+            Arg::with_name(DOMAIN_NAME)
+                .long(DOMAIN_NAME)
+                .value_name(DOMAIN_NAME)
+                .help("Domain's name as double-quoted string.")
+                .takes_value(true)
+                .required(true),
+        )
+    }
+
+    pub fn match_create(matches: &ArgMatches<'_>) {
+        if let Some(ref matches) = matches.subcommand_matches(DOMAIN) {
+            if let Some(domain_name) = matches.value_of(DOMAIN_NAME) {
+                println!("Creating domain with a name: {}", domain_name);
+                create_domain(domain_name);
+            }
+        }
+    }
+
+    fn create_domain(domain_name: &str) {
+        let create_domain = CreateDomain {
+            domain_name: String::from(domain_name),
+        };
+        let mut iroha_client = Client::new(
+            Configuration::from_path("config.json").expect("Failed to load configuration."),
```

Comment:
I think you have a cli argument that specifies the path to configuration file. Maybe it should be used here?

Reply:
It will be done in the next PR with publication of CLI tool as cargo plugin